CMSgov/design-system
# [WNMGDS-455] Fix webpack tree shaking

### Summary
- This PR adds ES modules to our `design-system` package to support Webpack 4's tree shaking
- The "Using React components" page has been rewritten and updated
- The `react-app` example has been updated and can be used to test tree shaking.

### How to test

1. Test the docs page
    - Check the file sizes of the doc page bundle and each example page.
    - Run `yarn build` to build the new ESM dist.
    - Check to see if files are significantly smaller.
2. Test with the `react-app` example
    - Change line 6 in the `package.json` in `react-app` to be: `"@cmsgov/design-system": "file:../../packages/design-system",`
    - Run `rm -rf node_modules` to delete node modules
    - Run `npm install` in the `react-app` example. 
    - Run `npm run build` to output the bundle.
    - Change line 4 of `react-app/src/scripts/index.js` to `import { Alert } from '@cmsgov/design-system/dist/index.js';` to simulate the original behavior, compare the bundle and ensure it is much larger
3. Check tree shaking is working in the child design systems [child-design-system-example.zip](https://github.com/CMSgov/design-system/files/4713248/child-design-system-example.zip)as well.
